app.go

```diff
@@ -0,0 +1,113 @@
+	a.DisablePermissions = &b
+	return a
+}
+
+func (a *AppSettings) SetAPNConfig(c APNConfig) *AppSettings {
+	a.APNConfig = &c
+	return a
+}
+
+func (a *AppSettings) SetFirebaseConfig(c FirebaseConfig) *AppSettings {
+	a.FirebaseConfig = &c
+	return a
+}
+
+func (a *AppSettings) SetWebhookURL(s string) *AppSettings {
+	a.WebhookURL = &s
+	return a
+}
+
+func NewAppSettings() *AppSettings {
+	return &AppSettings{}
+}
+
+type APNConfig struct {
+	Enabled              bool   `json:"enabled"`
+	Development          bool   `json:"development"`
+	AuthType             string `json:"auth_type,omitempty"`
+	AuthKey              []byte `json:"auth_key,omitempty"`
+	NotificationTemplate string `json:"notification_template"`
+	Host                 string `json:"host,omitempty"`
+	BundleID             string `json:"bundle_id,omitempty"`
+	TeamID               string `json:"team_id,omitempty"`
+	KeyID                string `json:"key_id,omitempty"`
+}
+
+type FirebaseConfig struct {
+	Enabled              bool   `json:"enabled"`
+	NotificationTemplate string `json:"notification_template"`
+}
+
+type PushNotificationFields struct {
+	APNConfig      APNConfig      `json:"apn"`
+	FirebaseConfig FirebaseConfig `json:"firebase"`
+}
+
+type Policy struct {
+	Name      string    `json:"name"`
+	Resources []string  `json:"resources"`
+	Roles     []string  `json:"roles"`
+	Action    int       `json:"action"` // allow: 1, deny: 0
+	Owner     bool      `json:"owner"`
+	Priority  int       `json:"priority"`
+	CreatedAt time.Time `json:"created_at"`
+	UpdatedAt time.Time `json:"updated_at"`
+}
+
+type AppResponseFields struct {
+	Name                 string                    `json:"name"`
+	OrganizationName     string                    `json:"organization"`
+	PushNotifications    PushNotificationFields    `json:"push_notifications"`
+	WebhookURL           string                    `json:"webhook_url"`
+	ConfigNameMap        map[string]*ChannelConfig `json:"channel_configs"`
+	Policies             map[string][]Policy       `json:"policies"`
+	Suspended            bool                      `json:"suspended"`
+	SuspendedExplanation string                    `json:"suspended_explanation"`
+	DisableAuth          bool                      `json:"disable_auth_checks"`
+	DisablePermissions   bool                      `json:"disable_permissions_checks"`
+}
+
+type appResponse struct {
+	App *AppResponseFields `json:"app"`
+}
+
+// GetApp returns app settings
+func (c *Client) GetApp() (*AppResponseFields, error) {
+	var resp appResponse
+
+	err := c.makeRequest(http.MethodGet, "app", nil, nil, &resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.App, nil
+}
+
+// UpdateAppSettings makes request to update app settings
+// Example of usage:
+//  settings := NewAppSettings().SetDisableAuth(true)
+//  err := client.UpdateAppSettings(settings)
+func (c *Client) UpdateAppSettings(settings *AppSettings) error {
```

Comment:
Update means you replace the settings with the provided object. With all that pointer use, I assume you're sending a patch here?

```go
// PatchAppConfig requests field changes for each non nil value.
func (c *Client) PatchAppConfig(patch  *AppConfigPatch) error {
```

Reply:
same as above

Reply:
Then document the behaviour.